cpp/include/raft/linalg/map_then_reduce.cuh

```diff
@@ -82,9 +86,64 @@ void mapThenReduce(OutType* out,
                    const InType* in,
                    Args... args)
 {
-  detail::mapThenReduceImpl<InType, OutType, MapOp, ReduceLambda, TPB, Args...>(
+  detail::mapThenReduceImpl<InType, OutType, IdxType, MapOp, ReduceLambda, TPB, Args...>(
     out, len, neutral, map, op, stream, in, args...);
 }
+
+/**
+ * @defgroup map_reduce Map-Reduce ops
+ * @{
+ */
+
+/**
+ * @brief CUDA version of map and then generic reduction operation
+ * @tparam InValueType the data-type of the input
+ * @tparam MapOp the device-lambda performing the actual map operation
+ * @tparam ReduceLambda the device-lambda performing the actual reduction
+ * @tparam IndexType the index type
+ * @tparam OutValueType the data-type of the output
+ * @tparam TPB threads-per-block in the final kernel launched
+ * @tparam Args additional parameters
+ * @param[in] handle raft::handle_t
+ * @param[in] in the input of type raft::device_vector_view
+ * @param[in] neutral The neutral element of the reduction operation. For example:
+ *    0 for sum, 1 for multiply, +Inf for Min, -Inf for Max
+ * @param[out] out the output reduced value assumed to be a raft::device_scalar_view
+ * @param[in] map the fused device-lambda
+ * @param[in] op the fused reduction device lambda
+ * @param[in] args additional input arrays
+ */
+template <typename InValueType,
+          typename MapOp,
+          typename ReduceLambda,
+          typename IndexType,
+          typename OutValueType,
+          int TPB = 256,
+          typename... Args>
+void map_reduce(const raft::handle_t& handle,
```

Comment:
Let's change the name of the file in that case and mark it a breaking change?

Reply:
Maybe instead of changing the name of the file and making it breaking, we could add a new file with the new name, just add the mdspan overload (not the one that accepts pointers) and deprecate this one?